src/sentry/models/identity.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import absolute_import
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import (
+    BoundedPositiveIntegerField, EncryptedJsonField, FlexibleForeignKey, Model
+)
+
+
+class IdentityProvider(Model):
+    """
+    An IdentityProvider is an instance of a provider.
+
+    For example, we may auto generate some, such as github.com, where it's
+    ``IdentityProvider(type='github', instance='github.com')``
+
+    When possible the instance should be the domain used.
+    """
+    __core__ = False
+
+    type = models.CharField(max_length=64)
+    instance = models.CharField(max_length=64)
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_identity'
+        unique_together = (('type', 'instance'),)
+
+
+# TODO(dcramer): pull in enum library
+class IdentityStatus(object):
+    UNKNOWN = 0
+    VALID = 1
+    INVALID = 2
+
+
+class Identity(Model):
```

Comment:
this needs 'scopes' added to the schema so we know what we have, and when we need to upgrade scopes